core/rawdb/accessors_indexes_test.go

```diff
@@ -17,15 +17,41 @@
 package rawdb
 
 import (
+	"hash"
 	"math/big"
 	"testing"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethdb"
 	"github.com/ethereum/go-ethereum/rlp"
+	"golang.org/x/crypto/sha3"
 )
 
+// testHasher is the helper tool for transaction/receipt list hashing.
+// The original hasher is trie, in order to get rid of import cycle,
+// use the testing hasher instead.
```

Comment:
Maybe I'm missing something -- but if you just want a hasher, which currently is in the `trie`, you could do what I did in this PR: https://github.com/ethereum/go-ethereum/pull/21232/files#diff-31c4aa3a4249d4755fc652d3e0087b98R58 -- move the `trie.keccakState` into `crypto.KeccakState`. There are all sorts of places where we could have a use for it. I

Reply:
No, this hasher is a fake version for `trie.Hasher` which we used to derive hash for LIST(transactions, receipts)

Reply:
It seems to me that the use of that hasher carries quite a large overhead, in terms of LOC changed. Could you explain why that is needed? I don't quite follow

Reply:
The `hasher` here is a customized tool which implements the `trie.Hasher` interface.

Originally we use `trie.Trie` to calculate a Merkle root for a list(e.g. transactions, receipts).
Here I need a fake hasher for testing purpose.

For `crypto.KeccakState ` it doesn't implement the interface I need.